Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -933,7 +933,41 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
 
         return resultMarker;
     }
-    // FIXME: Implement the other AXTextUnits.
+
+    // Paragraphs must be handled differently from word + sentence boundaries, as there is no paragraph break iterator.
+    // Rather, paragraph boundaries are based on rendered newlines and changes in nodes/renderers.
+    if (textUnit == AXTextUnit::Paragraph) {
+        size_t startLineIndex = currentRuns->lineID(runIndex).lineIndex;
+        unsigned offsetInStartLine = currentRuns->runLengthSumTo(startLineIndex - 1) - offset();
```

Comment:
Can we ever compute a `startLineIndex` of zero (thus making this `- 1` underflow)? Or is that impossible?

Reply:
Just updated and cleaned up this bit. Now, we just use the `startLineIndex` without subtracting 1, and compute the offset in a more correct way.